### PR TITLE
FreeRTOS Integration

### PR DESCRIPTION
**Pull request by JugalBili**
Added FreeRTOS Tasks/Threads for servo and logging [JIRA EN-1111]

Moved internal_size() function from protected to public such that buffer size can be accessed from pmic_board.ino

### CODE REVIEW

**Review by jmailloux — APPROVED** [review]

File: src/internal/circular_buffer.hpp [hunk 1]
```diff
@@ -85,6 +85,18 @@ class CircularBuffer
 		return tail_;
 	}
 
+	// size_t setHead(size_t head){
```
Comment (jmailloux) [inline src/internal/circular_buffer.hpp:88]: Don't need these.

File: src/SDFileLogger.h [hunk 2]
```diff
@@ -140,10 +163,44 @@ class SDFileLogger final : public LoggerBase
 		log_buffer_.reset();
 	}
 
+	void writeReadyBufferToSDFile()
```
Comment (jmailloux) [inline src/SDFileLogger.h:166]: If this is the exact same function as the writeBufferToSDFile(), why not just pass in the buffer to be written as a parameter?

**Review by jmailloux — CHANGES_REQUESTED** [review]

File: src/SDFileLogger.h (hunk 2 above)
Comment (JugalBili) [reply src/SDFileLogger.h]: Created template function for writeBufferToSDFile

File: src/internal/circular_buffer.hpp (hunk 1 above)
Comment (JugalBili) [reply src/internal/circular_buffer.hpp]: Removed

**Review by jmailloux — APPROVED** [review]

**Pull request merged**